### PR TITLE
Allow cert auth without token, add credential store TLS auth tests

### PR DESCRIPTION
**Pull request by honza-kasik**
Remove the premature null token check in HashicorpVaultCredentialStore that prevented TLS certificate authentication from working without a credential-reference. The VaultConnector already handles null/blank tokens by falling back to ClientCertificateLoginStrategy.

Replace the check with an early validation that fails only when neither a token nor a client certificate configuration is available, with a clear error message guiding the user.

Add HashicorpVaultCredentialStoreTlsAuthTestCase with 5 test scenarios covering cert auth with null token, whitespace tokens, full CRUD, untrusted cert, and missing cert.

### CODE REVIEW

File: src/main/java/org/wildfly/security/hashicorp/vault/_private/HashiCorpVaultLogger.java [hunk 1]
```diff
@@ -36,8 +36,8 @@ public interface HashiCorpVaultLogger extends BasicLogger {
     @Message(id = 2, value = "host-address attribute is required")
     CredentialStoreException hostAddressRequired();
 
-    @Message(id = 3, value = "Vault token is required")
-    CredentialStoreException vaultTokenRequired();
+    @Message(id = 3, value = "No authentication method configured: provide either a vault token (via credential-reference) or TLS client certificate authentication (via authentication-context)")
```
Comment (skyllarr) [inline src/main/java/org/wildfly/security/hashicorp/vault/_private/HashiCorpVaultLogger.java]: @honza-kasik Since this repo was already released, we shouldn't bechanging existing message IDs. Can you please add new message ID for this? Thanks!

File: src/test/java/org/wildfly/security/hashicorp/vault/HashicorpVaultCredentialStoreTestCase.java [hunk 2]
```diff
@@ -638,8 +638,10 @@ public void testInitializeWithBlankHostAddress() {
     }
 
     /**
-     * Call {@code initialize()} with {@code null} protection parameter (no token provided).
-     * Test passes when {@link CredentialStoreException} is thrown indicating token is required.
+     * Call {@code initialize()} with {@code null} protection parameter (no token provided) and no
+     * alternative auth method configured. The null token passes through to
+     * {@code VaultConnector.tryLoginWithFallback()} which fails because no login strategy succeeds.
```
Comment (skyllarr) [inline src/test/java/org/wildfly/security/hashicorp/vault/HashicorpVaultCredentialStoreTestCase.java]: IIUC this won't happen: `token passes through to
     * {@code VaultConnector.tryLoginWithFallback()}` thanks to early check in HashiCorpVaultCredentialStore which checks:

if (!hasToken && !hasClientCertConfig) {
                throw ROOT_LOGGER.noAuthenticationMethodConfigured();
            }

**Comment by skyllarr** [comment]
just some nitpicks otherwise LGTM! Thank you @honza-kasik !

File: src/main/java/org/wildfly/security/hashicorp/vault/_private/HashiCorpVaultLogger.java (hunk 1 above)
Comment (honza-kasik) [reply src/main/java/org/wildfly/security/hashicorp/vault/_private/HashiCorpVaultLogger.java]: Thanks, that makes sense, I reverted this change and added new ID instead.

File: src/test/java/org/wildfly/security/hashicorp/vault/HashicorpVaultCredentialStoreTestCase.java (hunk 2 above)
Comment (honza-kasik) [reply src/test/java/org/wildfly/security/hashicorp/vault/HashicorpVaultCredentialStoreTestCase.java]: Thank you! Nice catch! I got caught in the loop.

**Review by skyllarr — APPROVED** [review]
